pkg/api/norman/customization/cluster/validator.go

```diff
@@ -342,6 +346,127 @@ func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec
+
+	if !createFromImport {
+		if err := validateAKSKubernetesVersion(clusterSpec, prevCluster); err != nil {
+			return err
+		}
+		if err := validateAKSNodePools(clusterSpec); err != nil {
+			return err
+		}
+		if err := validateAKSAccess(request, aksConfig, prevCluster); err != nil {
+			return err
+		}
+	}
+
+	if request.Method != http.MethodPost {
+		return nil
+	}
+
+	return nil
+}
+
+// validateAKSCredentialAuth validates that a user has access to the credential they are setting and the credential
+// they are overwriting. If there is no previous credential such as during a create or the old credential cannot
+// be found, the auth check will succeed as long as the user can access the new credential.
+func validateAKSCredentialAuth(request *types.APIContext, credential string, prevCluster *v3.Cluster) error {
+	var accessCred mgmtclient.CloudCredential
+	credentialErr := "error accessing cloud credential"
+	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.CloudCredentialType, credential, &accessCred); err != nil {
+		return httperror.NewAPIError(httperror.NotFound, credentialErr)
+	}
+
+	if prevCluster == nil || prevCluster.Spec.AKSConfig == nil {
+		return nil
+	}
+
+	// validate the user has access to the old cloud credential before allowing them to change it
+	credential = prevCluster.Spec.AKSConfig.AzureCredentialSecret
+	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.CloudCredentialType, credential, &accessCred); err != nil {
+		if apiError, ok := err.(*httperror.APIError); ok {
+			if apiError.Code.Status == httperror.NotFound.Status {
+				// old cloud credential doesn't exist anymore, anyone can change it
+				return nil
+			}
+		}
+		return httperror.NewAPIError(httperror.NotFound, credentialErr)
+	}
+
+	return nil
+}
+
+// validateAKSKubernetesVersion checks whether a kubernetes version is provided
+func validateAKSKubernetesVersion(spec *v32.ClusterSpec, prevCluster *v3.Cluster) error {
+	clusterVersion := spec.AKSConfig.KubernetesVersion
+
+	if clusterVersion == nil {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "cluster kubernetes version cannot be empty string")
+	}
+
+	return nil
+}
+
+// validateAKSNodePools checks whether a given NodePool version is empty or not supported.
+// More involved validation is performed in the aks-operator.
+func validateAKSNodePools(spec *v32.ClusterSpec) error {
+	nodePools := spec.AKSConfig.NodePools
+	if nodePools == nil {
+		return nil
+	}
+	if len(nodePools) == 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one nodepool"))
+	}
+
+	var errors []string
+
+	for _, np := range nodePools {
+
+		name := np.Name
+		if name == nil && *name == "" {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("nodePool Name cannot be an empty"))
+		}
+
+		version := np.OrchestratorVersion
+		if version == nil {
```

Comment:
What about `*version == ""`?

Reply:
In aks-operator it's allowed to be empty https://github.com/rancher/aks-operator/pull/3/files#diff-1aa54d4941ca80708078867e1b130aff086719689479dea38683fded2f3c7852R516 should that be the case here? Or should aks-operator be changed?

Reply:
Yes, this can be removed